libsnark/gadgetlib1/tests/test_pairing.cpp

```diff
@@ -140,58 +139,83 @@ template<typename ppT> void test_mnt_miller_loop(const std::string &annotation)
     miller.generate_r1cs_witness();
     ASSERT_TRUE(pb.is_satisfied());
 
-    libff::affine_ate_G1_precomp<other_curve<ppT>> native_prec_P =
-        other_curve<ppT>::affine_ate_precompute_G1(P_val);
-    libff::affine_ate_G2_precomp<other_curve<ppT>> native_prec_Q =
-        other_curve<ppT>::affine_ate_precompute_G2(Q_val);
-    libff::Fqk<other_curve<ppT>> native_result =
-        other_curve<ppT>::affine_ate_miller_loop(native_prec_P, native_prec_Q);
-
     ASSERT_EQ(native_result, result.get_element());
     printf(
         "number of constraints for Miller loop (Fr is %s)  = %zu\n",
         annotation.c_str(),
         pb.num_constraints());
 }
 
-template<typename ppT>
-void test_mnt_e_over_e_miller_loop(const std::string &annotation)
+template<typename ppT> void test_mnt_miller_loop(const std::string &annotation)
 {
-    protoboard<libff::Fr<ppT>> pb;
-    libff::G1<other_curve<ppT>> P1_val =
+    libff::G1<other_curve<ppT>> P_val =
         libff::Fr<other_curve<ppT>>::random_element() *
         libff::G1<other_curve<ppT>>::one();
-    libff::G2<other_curve<ppT>> Q1_val =
+    libff::G2<other_curve<ppT>> Q_val =
         libff::Fr<other_curve<ppT>>::random_element() *
         libff::G2<other_curve<ppT>>::one();
 
-    libff::G1<other_curve<ppT>> P2_val =
+    libff::affine_ate_G1_precomp<other_curve<ppT>> native_prec_P =
+        other_curve<ppT>::affine_ate_precompute_G1(P_val);
+    libff::affine_ate_G2_precomp<other_curve<ppT>> native_prec_Q =
+        other_curve<ppT>::affine_ate_precompute_G2(Q_val);
+    libff::Fqk<other_curve<ppT>> native_result =
+        other_curve<ppT>::affine_ate_miller_loop(native_prec_P, native_prec_Q);
+
+    test_miller_loop<ppT>(P_val, Q_val, native_result, annotation);
+}
+
+template<typename ppT> void test_bls_miller_loop(const std::string &annotation)
+{
+    libff::G1<other_curve<ppT>> P_val =
         libff::Fr<other_curve<ppT>>::random_element() *
         libff::G1<other_curve<ppT>>::one();
-    libff::G2<other_curve<ppT>> Q2_val =
+    libff::G2<other_curve<ppT>> Q_val =
         libff::Fr<other_curve<ppT>>::random_element() *
         libff::G2<other_curve<ppT>>::one();
 
+    libff::G1_precomp<other_curve<ppT>> native_prec_P =
+        other_curve<ppT>::precompute_G1(P_val);
+    libff::G2_precomp<other_curve<ppT>> native_prec_Q =
+        other_curve<ppT>::precompute_G2(Q_val);
+    libff::Fqk<other_curve<ppT>> native_result =
+        other_curve<ppT>::miller_loop(native_prec_P, native_prec_Q);
+
+    test_miller_loop<ppT>(P_val, Q_val, native_result, annotation);
+}
+
+template<typename ppT>
+void test_e_over_e_miller_loop(
+    const libff::G1<other_curve<ppT>> &P1_val,
+    const libff::G2<other_curve<ppT>> &Q1_val,
+    const libff::G1<other_curve<ppT>> &P2_val,
+    const libff::G2<other_curve<ppT>> &Q2_val,
+    const FqkT<ppT> &expect_result,
```

Comment:
This is called `native_result` in the `test_miller_loop` function above. Let's use consistent names to represent the expected result here.

Reply:
Renamed to `expect_result`.

Reply:
Thanks